modules/tinymce/src/plugins/media/main/ts/core/Nodes.ts

```diff
@@ -5,26 +5,65 @@
  * For commercial licenses see https://www.tiny.cloud/
  */
 
+import { Arr, Obj, Type } from '@ephox/katamari';
 import Editor from 'tinymce/core/api/Editor';
 import Env from 'tinymce/core/api/Env';
+import DomParser from 'tinymce/core/api/html/DomParser';
 import AstNode from 'tinymce/core/api/html/Node';
 import * as Settings from '../api/Settings';
 import * as Sanitize from './Sanitize';
 import * as VideoScript from './VideoScript';
 
 declare let escape: any;
 
-const createPlaceholderNode = function (editor: Editor, node: AstNode) {
+const isLiveEmbedNode = (node: AstNode) => {
+  const name = node.name;
+  return name === 'iframe' || name === 'video' || name === 'audio';
+};
+
+const getDimension = (node: AstNode, styles: Record<string, string>, dimension: 'width' | 'height', defaultValue: string | null = null): string | null => {
```

Comment:
Should some of these nulls be Optionals?

Reply:
They could be, but we'd just have to unwrap them anyways since we're working with the public TinyMCE `Node` API and it requires `null` values.

Reply:
Another point worth mentioning, this is code being used for a node filter, so we generally try to avoid having to create unnecessary objects, function calls, etc... as it needs to be as performant as possible since content parsing happens a fair bit and can lead to bottlenecks (it's why core uses for loops instead of `Arr.each`).